### PR TITLE
Read only insight

### PR DESCRIPTION
**Pull request by jburel**
# What this PR does

Disable some features when the server is configured as read-only

Build on top of https://github.com/openmicroscopy/openmicroscopy/pull/5749
So we can test the feature exposed to the client

# Testing this PR
Check that 
* importer
* running script
* download
* creation of project/dataset/screen from top
* saving of settings
are disabled

### CODE REVIEW

File: components/insight/SRC/org/openmicroscopy/shoola/env/data/DataServicesFactory.java [hunk 1]
```diff
@@ -634,10 +633,12 @@ public void connect(UserCredentials uc)
         Set<GroupData> available;
         List<ExperimenterData> exps = new ArrayList<ExperimenterData>();
         String ldap = null;
+        long gid = exp.getDefaultGroup().getId();
+        SecurityContext ctx = new SecurityContext(gid);
+        boolean canCreate = omeroGateway.canCreate(ctx);
```
Comment (mtbc) [inline components/insight/SRC/org/openmicroscopy/shoola/env/data/DataServicesFactory.java:638]: Do you mean `Boolean.TRUE.equals(omeroGateway.canCreate(ctx))` to guard against nulls?

Comment (mtbc) [reply components/insight/SRC/org/openmicroscopy/shoola/env/data/DataServicesFactory.java:638]: Ah, sorry, maybe not, probably getting my gateways mixed up!

Comment (jburel) [reply components/insight/SRC/org/openmicroscopy/shoola/env/data/DataServicesFactory.java:638]: ``Null`` is already handled in the internal gateway so we have to deal in the rest of the code with  either ``true`` or ``false``

Comment (joshmoore) [reply components/insight/SRC/org/openmicroscopy/shoola/env/data/DataServicesFactory.java:638]: @jburel : if we start to take more conditions into account, how will this change in the future? Will you break the API?

Comment (jburel) [reply components/insight/SRC/org/openmicroscopy/shoola/env/data/DataServicesFactory.java:638]: This internal gateway is now in many cases a wrapper around the Java gateway under blitz.
If we introduce more conditions, I think we will be adding entry in the lookup. It won't break the API.

**Comment by pwalczysko** [comment]
missing bird's eye view on user-6 ID 21450


Edit: The same image works fine in iviewer. So I cannot guarantee that this is read-only problem, but I can guarantee that this is an insight specific problem

**Comment by pwalczysko** [comment]
Insight from https://ci.openmicroscopy.org/job/OMERO-DEV-merge-build/lastSuccessfulBuild/ICE=3.6,jdk=8_LATEST,label=octopus/ was used

**Comment by pwalczysko** [comment]
The menu of scripts from RHP is not disabled.

```
java.lang.Exception: java.lang.Exception: org.openmicroscopy.shoola.env.data.ProcessException: Cannot run script with ID:140162
	at org.openmicroscopy.shoola.env.data.views.ProcessBatchCall.doStep(ProcessBatchCall.java:101)
	at ...


```

screen shot 2018-05-04 at 12 17 13

**Comment by pwalczysko** [comment]
Create Report has not been disabled and is working. Expected ?

**Comment by pwalczysko** [comment]
Save thumbnails has not been disabled and is apparently working. Expected ?
screen shot 2018-05-04 at 12 21 13

**Comment by pwalczysko** [comment]
4 problems

   - https://github.com/openmicroscopy/openmicroscopy/pull/5752#issuecomment-386569644 Fixed see 
be7e8e1
   - https://github.com/openmicroscopy/openmicroscopy/pull/5752#issuecomment-386571288 Fixed see 6773631
   - https://github.com/openmicroscopy/openmicroscopy/pull/5752#issuecomment-386571776 Expected see below
   - https://github.com/openmicroscopy/openmicroscopy/pull/5752#issuecomment-386571906 Expected see below

**Comment by jburel** [comment]
``* Create Report has not been disabled and is working. Expected ?``
Yes this is done client side.
``* Save thumbnails has not been disabled and is apparently working. Expected ?``
Yes  this is done client side.

**Comment by pwalczysko** [comment]
Bug: The Delete key on keyboard still works, opens dialogs confirming the delete, then the activities spinner spins endlessly on read-only server (eel merge)

**Comment by pwalczysko** [comment]
Bug: Drag and drop on read-only in the tree (trying to D&D a Dataset into another Project) pops up a dialog (note, I am the owner of all the objects I am manipulating here)
screen shot 2018-05-04 at 13 58 46

**Comment by jburel** [comment]
Interestingly for the problem reported https://github.com/openmicroscopy/openmicroscopy/pull/5752#issuecomment-386569644
the thumbnail shows up then disappears
I double-checked in web and it is correctly visible in Web i.e. this is **not** a server problem cc @mtbc which is good.

**Comment by jburel** [comment]
@pwalczysko the problems reported should now all be fixed

**Comment by dominikl** [comment]
Looks good. Only additional thing I could think of is to add an error message to the OMERO.importer client, which says that you can't import on an read-only server directly after the log in and then exit. If you try now, the import fails, but doesn't say why (can't even view the exception for some reason). But can be done as separate PR.

**Comment by jburel** [comment]
@dominikl good point I did not check the importer standalone. I will do that in a follow up PR

**Comment by jburel** [comment]
#5749 has been merged

**Comment by jburel** [comment]
Thanks @dominikl and @pwalczysko for review.
Merging

**Pull request merged**